### ISSUE TITLE
Crash when tapping on disabled map annotations

### ISSUE THREAD

**Issue by johnnewman** [mapbox/mapbox-gl-native#3206]
In the Mapbox iOS SDK 3.0.0, tapping a map annotation that has its `MGLAnnotationImage`'s `enabled` property set to `NO` sometimes causes a crash.  Using the default of `YES` never crashes.  FWIW: we mix enabled and disabled annotations on the map, with the disabled ones being much smaller pin images.

Trace:
```objc
* thread #1: tid = 0x3beff1, 0x36750a6e libsystem_platform.dylib`_platform_memmove + 366, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x18e00000)
    frame #0: 0x36750a6e libsystem_platform.dylib`_platform_memmove + 366
    frame #1: 0x0166d832 Mapbox`-[MGLMapView handleSingleTapGesture:] + 2186
    frame #2: 0x28e6020a UIKit`_UIGestureRecognizerSendTargetActions + 142
    frame #3: 0x28aada82 UIKit`_UIGestureRecognizerSendActions + 170
    frame #4: 0x2893fda4 UIKit`-[UIGestureRecognizer _updateGestureWithEvent:buttonEvent:] + 1004
    frame #5: 0x28e61396 UIKit`___UIGestureRecognizerUpdate_block_invoke898 + 62
    frame #6: 0x28900afe UIKit`_UIGestureRecognizerRemoveObjectsFromArrayAndApplyBlocks + 298
    frame #7: 0x288fd098 UIKit`_UIGestureRecognizerUpdate + 2920
    frame #8: 0x2479c090 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 20
    frame #9: 0x2479a386 CoreFoundation`__CFRunLoopDoObservers + 282
    frame #10: 0x2479a7c4 CoreFoundation`__CFRunLoopRun + 972
    frame #11: 0x246ed0d8 CoreFoundation`CFRunLoopRunSpecific + 516
    frame #12: 0x246ececc CoreFoundation`CFRunLoopRunInMode + 108
    frame #13: 0x2da62af8 GraphicsServices`GSEventRunModal + 160
    frame #14: 0x289762dc UIKit`UIApplicationMain + 144
```

**Comment by 1ec5** [comment]
For what it's worth, this code is due to be rewritten as part of #3159. I haven't seen this crash in the OS X port, from which I intend to backport the new version.

**Comment by 1ec5** [comment]
This will be fixed in #3261.
